### PR TITLE
Option to lint a specified set of test_specs

### LINKED ISSUE #9392: Add option to select test specs for `pod lib lint`
Currently, it is only possible to test all of the test_specs or none with `--skip-tests`.  It would be nice to be able to choose which test specs to run.

For example, for a pod that has both Swift and Objective C tests, it would be useful to be able to only run the Objective C tests when `--use-libraries` is also set.

### PR DESCRIPTION
**Pull request by paulb777**
Fix #9392

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
   [Dimitris Koutsogiorgas](https://github.com/dnkoutso)
   [#9423](https://github.com/CocoaPods/CocoaPods/issues/9423)
 
+* Option to lint a specified set of test_specs
```
Comment (dnkoutso) [inline CHANGELOG.md]: should probably be an enhancement? If it is should it be against master instead?

File: lib/cocoapods/command/repo/push.rb [hunk 2]
```diff
@@ -31,6 +31,7 @@ def self.options
             ['--no-private', 'Lint includes checks that apply only to public repos'],
             ['--skip-import-validation', 'Lint skips validating that the pod can be imported'],
             ['--skip-tests', 'Lint skips building and running tests during validation'],
+            ['--test-specs=test-spec1,test-spec2,etc', 'List of test specs to run'],
```
Comment (dnkoutso) [inline lib/cocoapods/command/repo/push.rb]: should that be available for `push`? I'd suspect we want people to push with all tests ran or none. Maybe we can limit this on lib/spec `lint`?

File: CHANGELOG.md (hunk 1 above)
Comment (paulb777) [reply CHANGELOG.md]: It's arguable.  The case for it being a bugfix is that `pod lib lint` is unusable for testing some podspecs is some configurations without it.

Our use case is wanting to run unit tests for external PRs, but not integration tests that require secrets.

I'll go with whatever you think is best.

File: lib/cocoapods/command/repo/push.rb (hunk 2 above)
Comment (paulb777) [reply lib/cocoapods/command/repo/push.rb]: I'll revert this file.

File: CHANGELOG.md (hunk 1 above)
Comment (paulb777) [reply CHANGELOG.md]: Rebased and updated CHANGELOG

File: lib/cocoapods/validator.rb [hunk 3]
```diff
@@ -732,7 +736,17 @@ def test_pod
       else
         UI.message "\nTesting with `xcodebuild`.\n".yellow do
           pod_target = validation_pod_target
-          consumer.spec.test_specs.each do |test_spec|
+          all_test_specs = consumer.spec.test_specs
+          unless test_specs.nil?
+            test_spec_names = all_test_specs.map(&:base_name)
+            all_test_specs = all_test_specs.select { |test_spec| test_specs.include? test_spec.base_name }
```
[the remark this answers is not in the document]
Comment (paulb777) [reply lib/cocoapods/validator.rb]: Done.

**Comment by dnkoutso** [comment]
@paulb777 want to rebase for master (1.10)?

**Comment by paulb777** [comment]
@dnkoutso Rebased.

**Comment by dnkoutso** [comment]
@paulb777 sorry it needs one more rebase since 1.9.1 was released.

**Comment by paulb777** [comment]
@dnkoutso Rebased

**Pull request merged**